evalml/pipelines/utils.py

```diff
@@ -151,17 +169,79 @@ def _get_preprocessing_components(
             import_or_raise(
                 "imblearn.over_sampling", error_msg="imbalanced-learn is not installed"
             )
-            pp_components.append(sampler_components[sampler_name])
+            components.append(sampler_components[sampler_name])
         except ImportError:
             logger.warning(
                 "Could not import imblearn.over_sampling, so defaulting to use Undersampler"
             )
-            pp_components.append(Undersampler)
+            components.append(Undersampler)
+    return components
+
 
+def _get_standard_scaler(X, y, problem_type, estimator_class, sampler_name=None):
+    components = []
     if estimator_class and estimator_class.model_family == ModelFamily.LINEAR_MODEL:
-        pp_components.append(StandardScaler)
+        components.append(StandardScaler)
+    return components
+
+
+def _get_time_series_featurizer(X, y, problem_type, estimator_class, sampler_name=None):
+    components = []
+    if (
+        is_time_series(problem_type)
+        and estimator_class.model_family != ModelFamily.ARIMA
+    ):
+        components.append(DelayedFeatureTransformer)
+    return components
+
+
+def _get_preprocessing_components(
+    X, y, problem_type, estimator_class, sampler_name=None
+):
+    """Given input data, target data and an estimator class, construct a recommended preprocessing chain to be combined with the estimator and trained on the provided data.
+
+    Args:
+        X (pd.DataFrame): The input data of shape [n_samples, n_features].
+        y (pd.Series): The target data of length [n_samples].
+        problem_type (ProblemTypes or str): Problem type.
+        estimator_class (class): A class which subclasses Estimator estimator for pipeline.
+        sampler_name (str): The name of the sampler component to add to the pipeline. Defaults to None.
+
+    Returns:
+        list[Transformer]: A list of applicable preprocessing components to use with the estimator.
+    """
+    if is_time_series(problem_type):
+        components_functions = [
+            _get_label_encoder,
+            _get_drop_all_null,
+            _get_drop_index_unknown,
+            _get_url_email,
+            _get_natural_language,
+            _get_imputer,
+            _get_time_series_featurizer,
+            _get_datetime,
+            _get_ohe,
+            _get_sampler,
+            _get_standard_scaler,
+        ]
+    else:
+        components_functions = [
+            _get_label_encoder,
+            _get_drop_all_null,
+            _get_drop_index_unknown,
+            _get_url_email,
+            _get_datetime,
+            _get_natural_language,
+            _get_imputer,
+            _get_ohe,
+            _get_sampler,
+            _get_standard_scaler,
+        ]
+    components = []
+    for function in components_functions:
+        components.extend(function(X, y, problem_type, estimator_class, sampler_name))
```

Comment:
OK, I can get on board with this...it's clear in which circumstances you're generating the list in which order.